### DISCUSSION TITLE
How to update the openai package in the project to version 1.2.0?

### DISCUSSION THREAD

**Discussion by cadesalaberry** [chrislemke/ChatFred#124] [Q&A]
I was trying to update the ChatFred project to use the latest version of the openai pip package (to [use](https://platform.openai.com/docs/guides/images/usage?context=node) [dall-e 3](https://github.com/chrislemke/ChatFred/issues/120)), and I am not familiar with the way you packaged the dependencies for your project. I was expecting a `requirements.txt` file to update somewhere but I did not find it.

How were the dependencies initially shipped in the `libs` folder? How should I go about updating them? I thought about using `pip download` but it does not look like the expected file structure (it downloads `.whl` files).

**Comment by chrislemke** [comment]
Hey @cadesalaberry!
You are right, all packages are installed to the [libs folder](https://github.com/chrislemke/ChatFred/tree/main/workflow/src/libs). Just use `--target` with `pip install`. Hope this helps!

**Reply by cadesalaberry** [reply]
I think I am missing something, when I execute:
```
pip3 install --target workflow/src/libs openai --upgrade
```

![Screenshot 2023-11-10 at 11 39 37](https://github.com/chrislemke/ChatFred/assets/1447513/23456f9e-c5a6-46cf-ad24-0dd76f44ad44)


Here are my questions:
- I get `*.dist-info` folders
- Are the dependencies platform dependent? I am on M2, will it still work on intel Macs?

Is there another parameter that I am missing?

**Reply by chrislemke** [reply]
Hey! 
Usually, that should work for Apple silicon as well as x86. You can/should remove all `*.dist-info` as well as all `*.so` files (you can use [this script](https://github.com/chrislemke/ChatFred/blob/main/scripts/detect_banned_file_types.sh)) to detect them.

**Reply by cadesalaberry** [reply]
Thx! I have some more question to proceed with the update:
- How did you set the [python3 path](https://github.com/chrislemke/ChatFred/pull/125/files#diff-e929447fe0a2389c7a6b6869ad6ab52b98fe227948ed5a59f7c17721e7f5451eR1) should I leave it this way?
- To test the workflow, do you [package it](https://www.alfredforum.com/topic/12667-how-i-convert-a-folder-with-a-workflow-into-alfredworkflow/) into a `.alfredworkflow` and reinstall it every time? is there a faster way to test from a folder? also is there a way to backup preferences before installing a new workflow?

**Reply by chrislemke** [reply]
- There should be no need to change the python3 path
- For testing the easiest way is that you navigate to the workflow path and just exchange the files. No need for packaging it or for reinstallation.
